src/overlays/actors/ovl_Bg_Kin2_Bombwall/z_bg_kin2_bombwall.c

```diff
@@ -28,50 +40,181 @@ const ActorInit Bg_Kin2_Bombwall_InitVars = {
+            (Math3D_Vec3fDistSq(&arg0->dyna.actor.world.pos, &bombwallCollider->world.pos) < 6400.0f)) {
+            return 1;
+        }
+    }
+    return 0;
+}
+
+//first idea: with all the random numbers fetched, this is probably related to particle effects from slashing or blowing up
+void func_80B6E090(BgKin2Bombwall *this, GlobalContext *globalCtx) {
+    s32 i;
+    Vec3f pos; 
+    Vec3f velocity;
+    Vec3f spD8; 
+    Vec3f spCC;
+    s32 j;
+    s32 temp_s3;
+    f32 temp_a0;
+    s32 temp_s2;
+    s16 phi_s0;
+    s16 phi_s1;
+    
+
+    Matrix_RotateYS(this->dyna.actor.shape.rot.y, 0);
+    
+    for(i = 0, temp_s3 = 0; i < 6; i++) {
+        temp_a0 = (i + 1) * 15.f;   
+        for (j = 0; j < ARRAY_COUNT(D_80B6E730); j++) {
+            temp_s3++;
+            temp_s3 &= 7;
+            
+            spD8.x = D_80B6E730[j] + (s32)(((u32)Rand_Next()) >> 0x1C); 
+            spD8.y = ((Rand_ZeroOne() - 0.5f) * 15.0f) + temp_a0;  
+            spD8.z = (Rand_ZeroOne() * 20.0f) - 10.0f;
+            
+            spCC.x = (2.0f * (Rand_ZeroOne() - 0.5f)) + (spD8.x * 0.018461538f);
+            spCC.y = (Rand_ZeroOne() * 7.0f) + 4.0f;
+            spCC.z = spD8.z * 0.3f;
+            
+            Matrix_MultVec3f(&spD8, &pos);
+            Matrix_MultVec3f(&spCC, &velocity);
+    
+            pos.x += this->dyna.actor.world.pos.x;
+            pos.y += this->dyna.actor.world.pos.y;
+            pos.z += this->dyna.actor.world.pos.z;
+         
+            if (Rand_Next() % 4 == 0) {
+                phi_s0 = 0x20;
+            } else {
+                phi_s0 = 0x40;
+            }
+            
+            if (temp_s3 < 2 ||  Rand_Next() > 0) {
+                phi_s0 |= 1;
+                phi_s1 = 1;
+                func_800B0E48(globalCtx, &pos, &gZeroVec3f, &D_80B6E724, &D_80B6E71C, &D_80B6E720, 
+                                 (((u32) Rand_Next() >> 0x1B) + 0x46), 
+                                 (( ( (u32) Rand_Next() ) >> 0x1A) + 0x3C)); //for dust spawn
+            } else {
+                phi_s1 = 0;
+            }
+            EffectSsKakera_Spawn(globalCtx, &pos, &velocity, &pos, -0x226, phi_s0, 0x1E, 0, 
+                                 0,  D_80B6E738[temp_s3], phi_s1, 0, 0x32, -1, 
+                                 0x1F5,  D_06000128);
+        }
+    }
+}
+
+void BgKin2Bombwall_Init(Actor *thisx, GlobalContext *globalCtx) {
+    BgKin2Bombwall *this = (BgKin2Bombwall *) thisx;
+    ColliderCylinder *bombwallCollider;
+    
+    Actor_ProcessInitChain(&this->dyna.actor, D_80B6E748);
+    DynaPolyActor_Init((DynaPolyActor *) this, 0);
+    bombwallCollider = &this->collider;
+    Collider_InitCylinder(globalCtx, bombwallCollider);
+    if (Flags_GetSwitch(globalCtx, this->dyna.actor.params & 0x7F) != 0) {
+        Actor_MarkForDeath(&this->dyna.actor);
+        return;
+    }
+    DynaPolyActor_LoadMesh(globalCtx, (DynaPolyActor *) this, (CollisionHeader *) &D_06000490);
```

Comment:
```suggestion
    DynaPolyActor_LoadMesh(globalCtx, &this->dyna, &object_kin2_obj_Colheader_001328);
```
This data `D_06000490` should be in the object this actor uses, that object header should be imported and that data referenced instead of the externs (they were mostly a stop gap)
so add this to the top of this C file:
```
#include "objects/object_kin2_obj/object_kin2_obj.h"
```
I won't ask you to label/document object data... but the project leads might.